src/lib/core/coio.cc

```diff
@@ -177,34 +185,33 @@ coio_connect_timeout(const char *host, const char *service, int host_hint,
 			free(ai_local.ai_addr);
 	});
 	evio_timeout_update(loop(), &start, &delay);
-
 	coio_timeout_init(&start, &delay, timeout);
 	while (ai) {
-		try {
-			fd = coio_connect_addr(ai->ai_addr, ai->ai_addrlen,
-					       delay);
+		fd = coio_connect_addr(ai->ai_addr, ai->ai_addrlen, delay);
+		if (fd >= 0) {
 			if (addr != NULL) {
 				assert(addr_len != NULL);
 				*addr_len = MIN(ai->ai_addrlen, *addr_len);
 				memcpy(addr, ai->ai_addr, *addr_len);
 			}
 			return fd; /* connected */
-		} catch (SocketError *e) {
-			if (ai->ai_next == NULL)
-				throw;
-			/* ignore exception and try the next address */
 		}
+		if (ai->ai_next == NULL)
```

Comment:
Why do you return here? May be it's better to wait loop breaking in the next iteration (ai become equal to zero).

Reply:
Because the original code does, and I don't want to change it in the scope of this commit:

```diff
-               } catch (SocketError *e) {
-                       if (ai->ai_next == NULL)
-                               throw;
-                       /* ignore exception and try the next address */
```

I assume this is done to avoid `ev_now_update` call on the last iteration:

https://github.com/tarantool/tarantool/blob/b8bd3138f294cb08311f28b4908af7afc4041e65/src/lib/core/coio.cc#L197-L208